src/wallet/readable.go

```diff
@@ -16,11 +18,18 @@ type ReadableEntry struct {
 }
 
 // NewReadableEntry creates readable wallet entry
-func NewReadableEntry(w Entry) ReadableEntry {
+func NewReadableEntry(w Entry, isEncrypted bool) ReadableEntry {
+	var secret string
+	if !isEncrypted {
+		secret = w.Secret.Hex()
+	} else {
+		secret = w.EncryptedSeckey
+	}
+
 	return ReadableEntry{
 		Address: w.Address.String(),
 		Public:  w.Public.Hex(),
-		Secret:  w.Secret.Hex(),
+		Secret:  secret,
```

Comment:
Why not add `EncryptedSecret` to `ReadableEntry`?
The way here, is not easy to check if the value is encrypted or not

Reply:
The `Entry` should know if it is encrypted or not. The caller does not need to specify.  If `EncryptedSecret` is not empty then it is encrypted (and `Secret` should be empty).

Reply:
If it makes sense to leave `isEncrypted` argument, then check that `w.EncryptedSecret` is not empty

Reply:
There's a `encrypted` field in wallet meta info, the entries in the wallet would be either all encrypted or not encrypted. The `secret` field in `ReadableEntry` is string type, so we can decode it into `secret` or `encryptedSecret` in `Entry` struct base on the `encrypted` meta info in wallet,

Reply:
I am not sure we should have two different formats for the same field and toggle them based on a 3rd value

Reply:
Leave it as is for now, maybe entry can also have an encrypted bool field